crates/ty_python_semantic/resources/mdtest/scopes/nonlocal.md

```diff
@@ -43,3 +43,64 @@ def f():
         def h():
             reveal_type(x)  # revealed: Unknown | Literal[1]
 ```
+
+## The `nonlocal` keyword
+
+```py
+def f():
+    x = 1
+    def g():
+        x += 1  # error: [unresolved-reference]
+```
+
+Without the `nonlocal` keyword, `x += 1` (or `x = x + 1` or `x = foo(x)`) is not allowed in an inner
+scope like this. It might look like it would read the outer `x` and write to the inner `x`, but it
+actually tries to read the not-yet-initialized inner `x` and raises `UnboundLocalError` at runtime.
+
+```py
+def f():
+    x = 1
+    def g():
+        nonlocal x
+        x += 1
+```
+
+## TODO: get the write to land in the right scope
```

Comment:
From here on down, it's not currently passing. These are the bits I need help with.